pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -187,6 +187,26 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 		}
 	} else if err != nil {
 		return reconcile.Result{}, err
+	} else {
```

Comment:
I think this `else` could be removed (un-nest the contents), what do you think?

Reply:
I think no.

In that case, image check would be executed even when there was no deployment and just created in the first `if` block here https://github.com/aerogear/unifiedpush-operator/pull/8/files#diff-4deb70b03698dcadb6ba778ed52ce06bR183

Reply:
In that case it would be fine either with or without actually, right?

Thinking about when the deployment is created in the `if` block above: do you think maybe we should do a requeue after the create, like you saw in the memcached operator for other spec updates? :thinking:

Reply:
cc @pb82 any idea?

Reply:
@pb82 this could maybe have been an alternative solution to https://github.com/integr8ly/application-monitoring-operator/pull/38 also